Assets/ShooterProject/Scripts/General/GameObjectsPool.cs

```diff
@@ -0,0 +1,116 @@
+using System.Collections.Generic;
+using UnityEngine;
+namespace ShooterProject.Scripts.General
+{
+	public class GameObjectsPool
+	{
+		#region Fields
+
+		private List<GameObject> _objects;
+
+		private bool _autoExpand;
+		private bool _activeByDefault;
+		private Transform _container;
+		private GameObject _objectPrefab;
+		private bool _initialized;
+
+		#endregion
+
+		#region Properties
+
+		public int MaxSize { get; }
+
+		#endregion
+
+		#region Constructors
+		/// <summary>
+		/// Конструктор класса
+		/// </summary>
+		/// <param name="maxSize">Максимальный размер пула</param>
+		/// <param name="autoExpand">Использовать ли авторасширение</param>
+		/// <param name="activeByDefault">Параметр active объектов при инициализации или добавлении в пул</param>
+		/// <param name="prefab">Префаб объекта</param>
+		/// <param name="container">Контейнер пула</param>
+		public GameObjectsPool(int maxSize, bool autoExpand, bool activeByDefault, GameObject prefab, Transform container)
+		{
+			MaxSize = maxSize;
+			_autoExpand = autoExpand;
+			_activeByDefault = activeByDefault;
+			_objectPrefab = prefab;
+			_container = container;
+
+			InitPool();
+		}
+
+		#endregion
+
+		#region Public Methods
```

Comment:
Вставь пустую строку
```suggestion
		#region Public Methods
		
```